justfile

```diff
@@ -440,7 +451,62 @@ _linkerd-viz-uninit:
 
 
 # TODO linkerd-jaeger-install
-# TODO linkerd-multicluster-install
+
+##
+## linkerd multicluster
+## 
+
+mc-gateway-image := "gcr.io/google_containers/pause:3.2"
+
+linkerd-mc *flags: _k3d-init
+    {{ _linkerd }} mc {{ flags }}
+
+linkerd-mc-install: _linkerd-init linkerd-mc-load && _linkerd-mc-ready
+    {{ _linkerd }} mc install \
+            --set='linkerdVersion={{ linkerd-tag }}' \
+        | {{ _kubectl }} apply -f -
+
+# Wait for all test namespaces to be removed before uninstalling linkerd-multicluster.
+linkerd-mc-uninstall:
+    {{ _linkerd }} mc  uninstall \
+        | {{ _kubectl }} delete -f -
+
+linkerd-mc-build:
+	TAG={{ linkerd-tag }} bin/docker-build-controller
+
+linkerd-mc-load: _linkerd-mc-images _k3d-init
+	{{ _k3d-load }} \
+		{{ controller-image }}:{{ linkerd-tag }} \
+		{{mc-gateway-image }}
+
+_linkerd-mc-images:
+	#!/usr/bin/env bash
+	set -euo pipefail
+	for img in \
+		'{{ mc-gateway-image }}' \
+		'{{ controller-image }}:{{ linkerd-tag }}'
+	do
+		if [ -z $(docker image ls -q "$img") ]; then
+			exec {{ just_executable() }} \
+				linkerd-tag='{{ linkerd-tag }}' \
+				linkerd-mc-build
+		fi
+	done
```

Comment:
Given that we already build the controller image to install the core linkerd control plane and the core control plane has to be loaded on the cluster for any of this to work, do we really need these extra targets here? All we really need to do above and beyond the core linkerd loading is to ensure that the pause container is loaded _in the target cluster_.